<launch>
  <arg name="bag_path" description="bagfile path"/>
  <arg name="map_path" description="point cloud and lanelet2 map directory path"/>

  <arg name="use_sim_time" default="false" description="use_sim_time"/>
  <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
  <arg name="use_map_in_bag" default="false"/>

  <arg name="lanelet2_map_loader_param_path" default="$(find-pkg-share autoware_launch)/config/map/lanelet2_map_loader.param.yaml"/>
  <arg name="map_projection_loader_param_path" default="$(find-pkg-share autoware_launch)/config/map/map_projection_loader.param.yaml"/>

  <arg name="input/odometry" default="/localization/kinematic_state"/>
  <arg name="input/acceleration" default="/localization/acceleration"/>
  <arg name="input/trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="input/objects" default="/perception/object_recognition/objects"/>
  <arg name="input/map_topic" default="/map/vector_map"/>
  <arg name="input/route_topic" default="/planning/mission_planning/route"/>

  <group scoped="false">
    <include file="$(find-pkg-share global_parameter_loader)/launch/global_params.launch.py">
      <arg name="use_sim_time" value="$(var use_sim_time)"/>
      <arg name="vehicle_model" value="$(var vehicle_model)"/>
    </include>
  </group>

  <node_container pkg="rclcpp_components" exec="component_container" name="behavior_analyzer_container" namespace="" output="screen">
    <composable_node pkg="autoware_planning_data_analyzer" plugin="autoware::behavior_analyzer::BehaviorAnalyzerNode" name="behavior_analyzer">
      <param name="bag_path" value="$(var bag_path)"/>
      <param from="$(find-pkg-share autoware_planning_data_analyzer)/config/behavior_analyzer.param.yaml"/>
      <remap from="input/lanelet2_map" to="/map/vector_map"/>
    </composable_node>

    <!-- <composable_node pkg="map_loader" plugin="Lanelet2MapLoaderNode" name="lanelet2_map_loader"> -->
    <!--   <param from="$(var lanelet2_map_loader_param_path)"/> -->
    <!--   <param name="lanelet2_map_path" value="$(var map_path)/lanelet2_map.osm"/> -->
    <!--   <remap from="output/lanelet2_map" to="/map/vector_map"/> -->
    <!-- </composable_node> -->

    <composable_node pkg="glog_component" plugin="GlogComponent" name="glog_component" namespace=""/>
  </node_container>

  <!-- <include file="$(find-pkg-share autoware_map_projection_loader)/launch/map_projection_loader.launch.xml"> -->
  <!--   <arg name="param_path" value="$(var map_projection_loader_param_path)"/> -->
  <!--   <arg name="map_projector_info_path" value="$(var map_path)/map_projector_info.yaml"/> -->
  <!--   <arg name="lanelet2_map_path" value="$(var map_path)/lanelet2_map.osm"/> -->
  <!-- </include> -->
</launch>
